course6/toyapp-adam/math/vector.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+namespace Math {
+
+struct Vector2D
+{
+	float x = 0.0f;
+	float y = 0.0f;
+};
+
+struct Vector3D
+{
+	float x = 0.0f;
+	float y = 0.0f;
+	float z = 0.0f;
+
+	Vector3D operator-() const;
+	Vector3D operator+(const Vector3D& rhs) const;
+	Vector3D operator-(const Vector3D& rhs) const;
+	Vector3D operator*(float scalar) const;
+	Vector3D& operator+=(const Vector3D& rhs);
+	Vector3D& operator-=(const Vector3D& rhs);
+	Vector3D& operator*=(float scalar);
+};
+
+struct Vector4D
+{
+	float x;
+	float y;
+	float z;
+	float w;
```

Comment:
why do you not initilize them, while other vectors do initialize members?

Reply:
I initialize them in the constructor. I thought it's a bad idea to mix this kind of initialization with initializer lists if there are multiple constructors defined. Is it not?